pkg/ck8s/management_cluster.go

```diff
@@ -87,10 +88,25 @@ func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.O
 		return nil, &RemoteClusterConnectionError{Name: clusterKey.String(), Err: err}
 	}
 
+	authToken, err := token.Lookup(ctx, m.Client, clusterKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup auth token: %w", err)
+	}
+
+	if authToken == nil {
+		return nil, fmt.Errorf("auth token not yet generated")
+	}
+
+	if microclusterPort == 0 {
+		microclusterPort = 2380
+	}
```

Comment:
the more i see this, the more i think we should approach this differently.

Make the MicroclusterPort field a `*int`, then add a `GetMicroclusterPort()` method, which returns the field (if set), otherwise the default value (in this case 2380)`

then use `.GetMicroclusterPort()` for all invocations and remove this.

Reply:
done